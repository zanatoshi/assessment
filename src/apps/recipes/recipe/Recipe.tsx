import {makeStyles, Theme} from '@material-ui/core';
import React from 'react';
import {BreadcrumbsRecipe} from '../breadcrumbs/BreadcrumbsRecipe';
import {Ingredient, Ingredients} from './ingredients/Ingredients';
import {Instructions} from './instructions/Instructions';
import {Presentation} from './presentation/Presentation';

export interface Recipe {
  name: string;
  ingredients: Ingredient[];
  image: string;
  cookTime: string;
  instructions: string[];
  recipeYield: string;
  prepTime: string;
  description: string;
  id: string;
  totalTime: string;
  cost: string;
  meal: string;
  course: string;
}

interface RecipeProps {
  recipe: Recipe;
}

const useStyles = makeStyles((theme: Theme) => ({
  container: {
    padding: theme.spacing(1),
  },
  row: {
    display: 'flex',
    flexFlow: 'row',
    width: '100%',
    marginTop: theme.spacing(1),
  },
  breadcrumbs: {
    padding: theme.spacing(1),
  },
  picture: {
    width: '50%',
    padding: theme.spacing(1),
    '& img': {
      width: 'auto',
      height: 'auto',
      maxWidth: '100%',
    },
  },
}));

export const Recipe = ({recipe}: RecipeProps) => {
  const classes = useStyles();
  return (
    <div className={classes.container}>
      <div className={classes.breadcrumbs}>
        <BreadcrumbsRecipe course={recipe.course} meal={recipe.meal} />
      </div>
      <div className={classes.row}>
        <Presentation recipe={recipe} />
        <div className={classes.picture}>
          <img alt={recipe.name} src={recipe.image} />
        </div>
      </div>
      <hr />
      <div className={classes.row}>
        <Instructions instructions={recipe.instructions} />
        <Ingredients ingredients={recipe.ingredients} />
      </div>
    </div>
  );
};
