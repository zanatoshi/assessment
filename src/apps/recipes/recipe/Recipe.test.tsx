import {cleanup, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import {render} from '../../../test/test-utils';
import {Recipe} from './Recipe';

afterEach(() => cleanup());

describe('Recipe test', () => {
  const recipeTest: Recipe = {
    id: 'test',
    name: 'test-name',
    description: 'test-description',
    cookTime: 'test-10',
    prepTime: 'test-20',
    totalTime: 'test-30',
    course: 'main-course',
    meal: 'dinner',
    image: 'test.jpg',
    ingredients: [
      {ingredients: ['test-ingredient-1', 'test-ingredient-2'], line: 'use test-ingredient1&2'},
      {ingredients: ['test-ingredient-3'], line: 'use test-ingredient3'},
    ],
    instructions: ['test-instruction-1', 'test-instruction-2', 'test-instruction-3'],
    recipeYield: 'test makes',
    cost: 'test cost',
  };
  it('Show presentation, the ingredients and the instructions', () => {
    render(<Recipe recipe={recipeTest} />);
    expect(screen.getByText(recipeTest.name)).toBeTruthy();
    expect(screen.getByText(recipeTest.description)).toBeTruthy();
    expect(screen.getByText(recipeTest.meal.toUpperCase())).toBeTruthy();
    expect(screen.getByText(recipeTest.course.toUpperCase())).toBeTruthy();
    expect(screen.getByText(recipeTest.prepTime)).toBeTruthy();
    expect(screen.getByText(recipeTest.cookTime)).toBeTruthy();
    expect(screen.getByText(recipeTest.totalTime)).toBeTruthy();
    expect(screen.getByText(recipeTest.recipeYield)).toBeTruthy();
    recipeTest.ingredients.forEach((i) => {
      expect(screen.getByText(i.line)).toBeTruthy();
    });
    recipeTest.instructions.forEach((i) => {
      expect(screen.getByText(i)).toBeTruthy();
    });
  });

  it('Save recipe', async () => {
    render(<Recipe recipe={recipeTest} />);
    const saveButton = screen.getByText('SAVE RECIPE');
    expect(screen.queryByText('Should be saved !')).toBeNull();
    await userEvent.click(saveButton);
    expect(screen.getByText('Should be saved !')).toBeTruthy();
  });

  it('Print recipe', async () => {
    render(<Recipe recipe={recipeTest} />);
    const saveButton = screen.getByText('PRINT');
    expect(screen.queryByText('Should be printed !')).toBeNull();
    await userEvent.click(saveButton);
    expect(screen.getByText('Should be printed !')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const {asFragment} = render(<Recipe recipe={recipeTest} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
