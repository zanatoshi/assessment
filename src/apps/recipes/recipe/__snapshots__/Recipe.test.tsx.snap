// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-container"
  >
    <div
      class="makeStyles-breadcrumbs"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiBreadcrumbs-root makeStyles-breadcrumbs MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        <ol
          class="MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="/recipes"
            >
              RECIPES
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-separator MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="/recipes/meals/dinner"
            >
              DINNER
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-separator MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="/recipes/courses/main-course?meal=dinner"
            >
              MAIN-COURSE
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="makeStyles-row"
    >
      <div
        class="makeStyles-content"
      >
        <h2
          class="MuiTypography-root makeStyles-title MuiTypography-h2"
        >
          test-name
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
        >
          test-description
        </p>
        <div
          class="MuiBox-root MuiBox-root makeStyles-container"
        >
          <div
            class="MuiBox-root MuiBox-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon"
              display="block"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
              <path
                d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
              />
            </svg>
            <div
              class="MuiBox-root MuiBox-root"
            >
              <div
                class="MuiBox-root MuiBox-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  PREP
                </span>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1"
                >
                  test-20
                </h6>
              </div>
              <div
                class="MuiBox-root MuiBox-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  BAKE
                </span>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1"
                >
                  test-10
                </h6>
              </div>
              <div
                class="MuiBox-root MuiBox-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  TOTAL
                </span>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1"
                >
                  test-30
                </h6>
              </div>
            </div>
          </div>
        </div>
        <hr />
        <div
          class="MuiBox-root MuiBox-root makeStyles-container"
        >
          <div
            class="MuiBox-root MuiBox-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon"
              display="block"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
              <circle
                cx="8"
                cy="14"
                r="2"
              />
              <circle
                cx="12"
                cy="8"
                r="2"
              />
              <circle
                cx="16"
                cy="14"
                r="2"
              />
            </svg>
            <div
              class="MuiBox-root MuiBox-root"
            >
              <div
                class="MuiBox-root MuiBox-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  Yield
                </span>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1"
                >
                  test makes
                </h6>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-action"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                  SAVE RECIPE
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-action"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 8H5c-1.66 0-3 1.34-3 3v6h4v4h12v-4h4v-6c0-1.66-1.34-3-3-3zm-3 11H8v-5h8v5zm3-7c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm-1-9H6v4h12V3z"
                      />
                    </svg>
                  </span>
                  PRINT
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-picture"
      >
        <img
          alt="test-name"
          src="test.jpg"
        />
      </div>
    </div>
    <hr />
    <div
      class="makeStyles-row"
    >
      <div
        class="makeStyles-instructions"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          Instructions
        </h4>
        <ul
          class="MuiTimeline-root MuiTimeline-alignLeft makeStyles-timelineInstructions"
        >
          <li
            class="MuiTimelineItem-root MuiTimelineItem-alignLeft MuiTimelineItem-missingOppositeContent"
          >
            <div
              class="MuiTimelineSeparator-root"
            >
              <span
                class="MuiTimelineDot-root MuiTimelineDot-outlinedSecondary"
              />
              <span
                class="MuiTimelineConnector-root"
              />
            </div>
            <div
              class="MuiTimelineContent-root MuiTimelineItem-content"
            >
              <div
                class="MuiPaper-root makeStyles-instruction MuiPaper-elevation3 MuiPaper-rounded"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  test-instruction-1
                </p>
              </div>
            </div>
          </li>
          <li
            class="MuiTimelineItem-root MuiTimelineItem-alignLeft MuiTimelineItem-missingOppositeContent"
          >
            <div
              class="MuiTimelineSeparator-root"
            >
              <span
                class="MuiTimelineDot-root MuiTimelineDot-outlinedSecondary"
              />
              <span
                class="MuiTimelineConnector-root"
              />
            </div>
            <div
              class="MuiTimelineContent-root MuiTimelineItem-content"
            >
              <div
                class="MuiPaper-root makeStyles-instruction MuiPaper-elevation3 MuiPaper-rounded"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  test-instruction-2
                </p>
              </div>
            </div>
          </li>
          <li
            class="MuiTimelineItem-root MuiTimelineItem-alignLeft MuiTimelineItem-missingOppositeContent"
          >
            <div
              class="MuiTimelineSeparator-root"
            >
              <span
                class="MuiTimelineDot-root MuiTimelineDot-outlinedSecondary"
              />
              <span
                class="MuiTimelineConnector-root"
              />
            </div>
            <div
              class="MuiTimelineContent-root MuiTimelineItem-content"
            >
              <div
                class="MuiPaper-root makeStyles-instruction MuiPaper-elevation3 MuiPaper-rounded"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  test-instruction-3
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="makeStyles-ingredients"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        >
          Ingredients
        </h4>
        <div>
          <hr />
          <div
            class="makeStyles-ingredient"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              use test-ingredient1&2
            </p>
          </div>
        </div>
        <div>
          <hr />
          <div
            class="makeStyles-ingredient"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              use test-ingredient3
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
